helm/templates/deployment.yaml

```diff
@@ -66,10 +64,31 @@ spec:
                   # If 'false", the secret must exist before the pod can start
                   # App versions before v2.3.4 will read this value from a config file, not the env var.
                   optional: true
+            - name: RABBITMQ_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "idx.rabbitmq.secretPasswordName" . }}
+                  key: rabbitmq-password
+                  optional: false
+            - name: SOLR_ADMIN_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "idx.solr.secretPasswordName" . }}
+                  key: SOLR_ADMIN_PASSWORD
+                  optional: false
       initContainers:
         - name: init-solr
           image: busybox:latest
-          command: ['sh', '-c', "until nc -z {{ .Values.solr.hostname}} {{ .Values.solr.service.ports.http }} > /dev/null; do echo waiting for Solr; sleep 5; done"]
+          command:
+            - sh
+            - -c
+            - >
+              until [[ $(wget -O- "{{- .Values.solr.hostname -}}:
+              {{- .Values.solr.service.ports.http -}}/solr/
+              {{- .Values.solr.customCollection -}}
+              /admin/file?file=schema.xml&contentType=text/xml"{{- printf " " -}}
+              | grep -c "<schema name=\"dataone\"") == 1 ]]; do
+                echo waiting for Solr Schema to be accessible at ; sleep 1; done
```

Comment:
`nc` is pretty universally present on images, which is why initcontainers tend to use it for networ checks. Its also both lightweight and highly performant. `wget` is more of an end-user app. If you need it, we need to be sure it is a dependency in the underlying Docker image, which isn't always the case.

Reply:
the original check, using `nc`, only checks if the port is open. It doesn't verify that the schema doc is available. We were seeing indexer startup errors because the indexer code tries to get the schema as soon as the `nc` call is successful, but fails, presumably because solr hasn't fully started up.

afaict, there's no way to verify the schema's available using `nc` (correct me if I'm wrong). I would have preferred `curl`, but it is not available by default on `busybox` and `wget` is. If there's a different command that is available on busybox and will do a better job, I'm happy to switch.

Reply:
yeah, it's fine, like I said we just need to be sure wget is available. nc can do any network transfer across a port, but its very low-level. You have to send the HTTP request and close the connection by talking http (or whatever protocol is supported on the port). E.g., 

```bash
❯ nc -c arcticdata.io 80
GET / HTTP/1.1
Host: arcticdata.io
Connection: close

HTTP/1.1 302 Found
Date: Thu, 02 Nov 2023 15:13:26 GMT
Server: Apache
Location: https://arcticdata.io/
Content-Length: 206
Connection: close
Content-Type: text/html; charset=iso-8859-1

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="https://arcticdata.io/">here</a>.</p>
</body></html>
```

Reply:
Thanks for the edification - I figured out how to do it with `nc` :-)